Add policy for membership (#206)
jupyterlab/team-compass: docs/team/becoming-member.md

# Becoming a JupyterLab team member

> This document describe first how to become a council member. Then the two subgroups of JupyterLab Council member will be specified:

Reviewer: ```suggestion
This document describes how to become a council member. It includes information about the two subgroups of JupyterLab Council members:
```

> This document describe first how to become a council member. Then the two subgroups of JupyterLab Council member will be specified:

Reviewer 2: ```suggestion
This document describe first how to become a council member. Then the two subgroups of JupyterLab Council members will be specified:
```

Author: Superseeded by Jason W. suggestion

> For example, a member who is going out on a long leave/vacation (>2 weeks) can temporarily change their status to inactive during their absence and immediately reactivate upon return. This isn't required, but this can relieve them from having to watch this repository for any formal votes that happen during their absence.

Author: I removed the notion of inactive member as it adds complexity instead of clarity as pointed out by Brian.

   if they are interested. Don't forget to run them by the {ref}`membership_guidelines`
   page to make sure they understand what they're signing up for.

### Membership maintenance

> Every six months, a bot will open an issue in the [council repo](https://github.com/jupyterlab/council) asking all currently active JupyterLab council team members to reply within three weeks if they still consider themselves active. If a team member replies no, they will be removed from the council. If a member does not reply, the council may reach that member personnaly for membership confirmation. If that request for confirmation is not answer within a month, the member will be removed from the council.

Reviewer 3: ```suggestion
Every six months, a bot will open an issue in the [council repo](https://github.com/jupyterlab/council) asking all currently active JupyterLab council team members to reply within three weeks if they still consider themselves active. If a team member replies no, they will be removed from the council. If a member does not reply, the council may reach that member personnaly for membership confirmation. If that request for confirmation is not answered within a month, the member will be removed from the council.
```

Author: thanks

   if they are interested. Don't forget to run them by the {ref}`membership_guidelines`
   page to make sure they understand what they're signing up for.

### Membership Maintenance

> Every six months, one currently active member should open an issue in the team-compass repo asking all currently active team members to reply within 3 weeks if they still consider themselves active. If not (or no response is given by a team member), it will be assumed that they have gone inactive. This will help keep the active member list up-to-date.

Reviewer: ```suggestion
Every six months, one currently active member should open an issue in the team-compass repo asking all currently active team members to reply within three weeks if they still consider themselves active. If a team member replies no, or does not reply, we will conclude that they are inactive. This will help keep the active member list up to date.
```

Remember, an inactive member can return at any time by simply changing their status on the team-compass page.

## Release team member

### Joining the team

> Any JupyterLab Council member can ask to be part of the release team using the weekly call or the chat.

Reviewer 2: ```suggestion
Any JupyterLab Council member can ask to be part of the release team using the weekly call, chat, or official council email list.
```

### Release team responsibilities

> Release team member are granting the following rights:

Reviewer: ```suggestion
Release team members can manage the following teams:
```

- GitHub JupyterLab organization: membership of the [release team](https://github.com/orgs/jupyterlab/teams/release)
- PyPI Jupyter organization: manager of the [JupyterLab team](https://pypi.org/manage/organization/jupyter/team/jupyterlab)

> - NPM jupyterlab organization: membership of the [jupyterlab team](https://www.npmjs.com/settings/jupyterlab/teams)

Reviewer 2: conda-forge packages as well?

Author: This is a good point. I think the right on JupyterLab package should be granted (pushing on all Python packages that matches PyPI organization may be complex).

> - NPM jupyterlab organization: membership of the [jupyterlab team](https://www.npmjs.com/settings/jupyterlab/teams)

Author: ```suggestion
- NPM jupyterlab organization: membership of the [jupyterlab team](https://www.npmjs.com/settings/jupyterlab/teams)
- conda-forge jupyterlab recipe: maintainer rights
```

> > One of the admin will need to grant the new rights on all three web services.

Reviewer: ```suggestion
> An admin will need to add a new member to all three teams.
```

> > One of the admin will need to grant the new rights on all three web services.

Reviewer 2: ```suggestion
> One of the admin team members will need to grant the new rights on all three web services.
```

Author: Superseeded by Jason suggestion

> Being part of those teams, you will be granted publication rights of major JupyterLab-related packages. Meaning you should be able to carry out a release and react quickly in case a published package is broken or corrupted.

Reviewer: ```suggestion
Members of these teams have publication rights for major JupyterLab-related packages. Team members can publish a release and can react quickly in case a published package is broken or corrupted.
```

> Being part of those teams, you will be granted publication rights of major JupyterLab-related packages. Meaning you should be able to carry out a release and react quickly in case a published package is broken or corrupted.

Reviewer 2: ```suggestion
Being part of those teams, you will be granted publication rights of major JupyterLab-related packages. You should be able to carry out a release and react quickly in case a published package is broken or corrupted.
```

Author: Superseeded by Jason W. suggestion

### Membership Maintenance

> Every six months, a bot will open an issue in the team-compass repo asking all currently team members to reply within 3 weeks if they still consider themselves active. If not (or no response is given by a team member), it will be assumed that they have gone inactive. This will imply the member will be removed for this subteam.

Reviewer: ```suggestion
Every six months, a bot will open an issue in the team-compass repo asking all currently team members to reply within three weeks if they still consider themselves active. If a member replies no, or they do not reply, we will conclude that they are inactive. Inactive members may be removed from their teams.
```

> Every six months, a bot will open an issue in the team-compass repo asking all currently team members to reply within 3 weeks if they still consider themselves active. If not (or no response is given by a team member), it will be assumed that they have gone inactive. This will imply the member will be removed for this subteam.

Reviewer 2: ```suggestion
Every six months, a bot will open an issue in the team-compass repo asking all currently team members to reply within 3 weeks if they still consider themselves active. If not (or no response is given by a team member), it will be assumed that they have gone inactive. This will imply the member will be removed for this team.
```

Author: Superseeded by Jason W. suggestion

## Admin team member

> ### Joining the admin

Reviewer: ```suggestion
### Joining the admin team
```

> Any JupyterLab Council member can ask to be part of the admin team using the weekly call or the chat. If the number of administrators has reached 4 people; consensus should be achieved between the interestee and the existing administrators to get the new 4-people team.

Reviewer 2: ```suggestion
Any JupyterLab Council member can ask to be part of the admin team using the weekly call, chat, or official council email list. If the number of administrators has reached 4 people, consensus should be achieved between the interestee and the existing administrators to get the new 4-person team.
```

Author: Merged with Jason W. suggestion

> Any JupyterLab Council member can ask to be part of the admin team using the weekly call or the chat. If the number of administrators has reached 4 people; consensus should be achieved between the interestee and the existing administrators to get the new 4-people team.

Reviewer: What is "the chat"?

```suggestion
Any JupyterLab Council member can ask to be part of the admin team using the weekly call or the chat. There should be no more than four administrators. If there are four administrators already, consensus should be achieved between the new admin candidate and the existing administrators to determine a new four-person admin team.
```

> Any JupyterLab Council member can ask to be part of the admin team using the weekly call or the chat. If the number of administrators has reached 4 people; consensus should be achieved between the interestee and the existing administrators to get the new 4-people team.

Reviewer 2: I would suggest 6 people as the limit rather than 4, to make sure that there is someone available when it is needed, and to add some members of the EC, for example, as a backstop.

> > The Software Steering Council representative will automatically be part of this group in addition to the 4 members.

Reviewer: ```suggestion
> The Software Steering Council representative will automatically be part of this group, in addition to the four members chosen above.
```

### Admin team responsibilities

> Admin team member are granting the following rights:

Reviewer 2: ```suggestion
Admin team members are granted the following rights:
```

Author: Superseeded by Jason W. suggestion

> Admin team member are granting the following rights:

Reviewer: ```suggestion
Admin team members manage the following teams:
```

- GitHub JupyterLab organization: [ownership](https://github.com/orgs/jupyterlab)
- PyPI Jupyter organization: manager of the [JupyterLab team](https://pypi.org/manage/organization/jupyter/team/jupyterlab)

> - NPM jupyterlab organization: [ownership](https://www.npmjs.com/settings/jupyterlab)

Reviewer 2: conda-forge packages as well?

> Being administrators, you will have to maintain the rights of the member on the different projects.

Reviewer 2: ```suggestion
Being administrators, you will have to maintain the rights of the members on the different projects.
```

Author: Superseeded by Jason W suggestion

> Being administrators, you will have to maintain the rights of the member on the different projects.

Reviewer: ```suggestion
Administrators maintain membership of the teams listed above.
```

### Membership Maintenance

> Every six months, a bot will open an issue in the team-compass repo asking all currently team members to reply within 3 weeks if they still consider themselves active. If not (or no response is given by a team member), it will be assumed that they have gone inactive. This will imply the member will be removed for this subteam.

Reviewer 2: ```suggestion
Every six months, a bot will open an issue in the team-compass repo asking all currently team members to reply within 3 weeks if they still consider themselves active. If not (or no response is given by a team member), it will be assumed that they have gone inactive. This will imply the member will be removed from this team.
```

Author: Superseeded by Jason W. suggestion

> Every six months, a bot will open an issue in the team-compass repo asking all currently team members to reply within 3 weeks if they still consider themselves active. If not (or no response is given by a team member), it will be assumed that they have gone inactive. This will imply the member will be removed for this subteam.

Reviewer: ```suggestion
Every six months, a bot will open an issue in the team-compass repo asking all currently team members to reply within three weeks if they still consider themselves active. If a member says no, or they do not respond, we will conclude that they are inactive. Inactive members may be removed from their teams.
```

   if they are interested. Don't forget to run them by the {ref}`membership_guidelines`
   page to make sure they understand what they're signing up for.

### Membership Maintenance

> Every six months, one currently active member should open an issue in the team-compass repo asking all currently active team members to reply within three weeks if they still consider themselves active. If a team member replies no, or does not reply, we will conclude that they are inactive. This will help keep the active member list up to date.

Author: ```suggestion
Every six months, a bot will open an issue in the team-compass repo asking all currently active team members to reply within three weeks if they still consider themselves active. If a team member replies no, or does not reply, we will conclude that they are inactive. This will help keep the active member list up to date.
```

Remember, an inactive member can return at any time by simply changing their status on the team-compass page.

## Release team member

### Joining the team

Any JupyterLab Council member can ask to be part of the release team using the weekly call, chat, or official council email list.

### Release team responsibilities

Release team members can manage the following teams:

- GitHub JupyterLab organization: membership of the [release team](https://github.com/orgs/jupyterlab/teams/release)
- PyPI Jupyter organization: manager of the [JupyterLab team](https://pypi.org/manage/organization/jupyter/team/jupyterlab)
- NPM jupyterlab organization: membership of the [jupyterlab team](https://www.npmjs.com/settings/jupyterlab/teams)
- conda-forge jupyterlab recipe: maintainer rights

> An admin will need to add a new member to all three teams.

> Members of these teams have publication rights for major JupyterLab-related packages. Team members can publish a release and can react quickly in case a published package is broken or corrupted.

Reviewer 4: Not sure what "these teams" refers to. Also, this makes it sound like the release team are the only folks who can cut releases. But the language above suggests that the release team only manages the group of people who can cut releases (which is a separate team). Part of the reason we should clarify this is that our different repos in the JupyterLab org often have different groups of people who release them.

Author: Wording update for clarification.

I did not mention an explicit list of packages. But the pratical consequence of what is described here is that release team member will be able to release `jupyterlab` but also `jupyterlab-git` or the Jupyter renderers as those packages are officially supported as JupyterLab extensions. But it is true that there is a gray area for repository such as [jupyterlab-latex](https://github.com/jupyterlab/jupyterlab-latex), [jupyter-ai](https://github.com/jupyterlab/jupyter-ai) or [jupyterlab-hdf5](https://github.com/jupyterlab/jupyterlab-hdf5).

### Membership Maintenance

> Every six months, a bot will open an issue in the team-compass repo asking all currently team members to reply within three weeks if they still consider themselves active. If a member replies no, or they do not reply, we will conclude that they are inactive. Inactive members may be removed from their teams.

Reviewer 4: Because we have similar language for other roles, it would be helpful to be specific about which role the current paragraph applies. This could be addressed by changing "current team members" to "current release team members".

Author: Clarify this point.

## Admin team member

### Joining the admin team

Any JupyterLab Council member can ask to be part of the admin team using the weekly call, chat, or official council email list. There should be no more than four administrators. If there are four administrators already, consensus should be achieved between the new admin candidate and the existing administrators to determine a new four-person admin team.

> The Software Steering Council representative will automatically be part of this group, in addition to the four members chosen above.

### Admin team responsibilities

Admin team members manage the following teams:

- GitHub JupyterLab organization: [ownership](https://github.com/orgs/jupyterlab)
- PyPI Jupyter organization: manager of the [JupyterLab team](https://pypi.org/manage/organization/jupyter/team/jupyterlab)
- NPM jupyterlab organization: [ownership](https://www.npmjs.com/settings/jupyterlab)
- conda-forge jupyterlab recipe: maintainer rights

Administrators maintain membership of the teams listed above.

### Membership Maintenance

> Every six months, a bot will open an issue in the team-compass repo asking all currently team members to reply within three weeks if they still consider themselves active. If a member says no, or they do not respond, we will conclude that they are inactive. Inactive members may be removed from their teams.

Reviewer 4: "all current administrator team members"

Author: :+1:
